### ISSUE TITLE
RFE: Add a way to return whether a module build requires or requires a particular platform

### ISSUE THREAD

**Issue by mohanboddu** [fedora-modularity/libmodulemd#173]
As part of mass rebuild we need to know what modules should be rebuilt and that can be identified by looking at the the modulemd build requires, it would be easy if libmodulemd can provide that info when a modulemd with a platform is provided.

Same as above but during branching, we need to identify the modules that runtime requires a certain platform. So, when a modulemd is provided with a platform, please return whether the platform is a runtime requirement or not.

**Comment by sgallagher** [comment]
@mohanboddu Would an API like this work?

```python
#!/usr/bin/python3

import gi
gi.require_version('Modulemd', '2.0')
from gi.repository import Modulemd

index = Modulemd.ModuleIndex.new()
index.update_from_file('./modules.yaml', False)

modules_to_build = index.get_modules_that_build_depend_on_stream ('platform', '31')
```

The `modules_to_build` would be a list of `Modulemd.ModuleStream` objects that matched the provided module and stream in buildrequires. The same would be offered for requires also (as `index.get_modules_that_depend_on_stream ('platform', '31')`)

Also pinging @puiterwijk for his opinion.

I think I'll publish this as `Stability: unstable` for now.

**Comment by sgallagher** [comment]
@mohanboddu In the specific case of the mass-rebuild that we discussed in-person yesterday, you'd create a new `index` for each yaml document in the dist-git repos and then you could just check `size(modules_to_build)` to determine if that build should be included (since there would only be one module in the index, so either it will or won't be returned).

**Comment by mohanboddu** [comment]
> @mohanboddu Would an API like this work?
> 
> ```python
> #!/usr/bin/python3
> 
> import gi
> gi.require_version('Modulemd', '2.0')
> from gi.repository import Modulemd
> 
> index = Modulemd.ModuleIndex.new()
> index.update_from_file('./modules.yaml', False)
> 
> modules_to_build = index.get_modules_that_build_depend_on_stream ('platform', '31')
> ```
> 
> The `modules_to_build` would be a list of `Modulemd.ModuleStream` objects that matched the provided module and stream in buildrequires. The same would be offered for requires also (as `index.get_modules_that_depend_on_stream ('platform', '31')`)
> 
> Also pinging @puiterwijk for his opinion.
> 
> I think I'll publish this as `Stability: unstable` for now.

@sgallagher I think it will work.

Thanks.

**Comment by sgallagher** [comment]
@mohanboddu While implementing, I've realized that the above approach won't work (mostly because the contents in dist-git have no `name` or `version` fields, because that's populated by MBS from the dist-git repo and branch. The new approach looks like this:

```python3
import gi
gi.require_version('Modulemd', '2.0')
from gi.repository import Modulemd
stream = Modulemd.ModuleStream.read_file('something.yaml', True)

# Run-time requires
stream.depends_on_stream('platform', 'f30')

# Build-time requires
stream.build_depends_on_stream('platform', 'f30')
```

**Comment by mohanboddu** [comment]
@sgallagher That should work.

Thanks.

**Comment by sgallagher** [comment]
@mohanboddu I've pushed the fix and build libmodulemd 2.1.0 for Rawhide, F29 and F28. You can see the API documentation for the new functions at https://sgallagh.fedorapeople.org/docs/libmodulemd/2.0/modulemd-2.0-ModulemdModuleStream.html#modulemd-module-stream-depends-on-stream
